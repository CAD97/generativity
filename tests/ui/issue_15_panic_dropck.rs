use generativity::{Id, make_guard};

fn assert_eq_lt<'id>(_: Id<'id>, _: Id<'id>) {}

fn main() {
    make_guard!(g_a);
    make_guard!(g_b);

    let a: Id = g_a.into();
    let b: Id = g_b.into();

    assert_eq_lt(a, b);

    loop {}
}
