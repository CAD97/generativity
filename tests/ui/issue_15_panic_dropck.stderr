error[E0716]: temporary value dropped while borrowed
  --> tests/ui/issue_15_panic_dropck.rs:7:5
   |
7  |     make_guard!(g_b);
   |     ^^^^^^^^^^^^^^^^ creates a temporary value which is freed while still in use
...
15 | }
   | -
   | |
   | temporary value is freed at the end of this statement
   | borrow might be used here, when `lifetime_brand` is dropped and runs the `Drop` code for type `LifetimeBrand`
   |
   = note: consider using a `let` binding to create a longer lived value
   = note: this error originates in the macro `make_guard` (in Nightly builds, run with -Z macro-backtrace for more info)
